lib/ViewModels/CatalogItemNameSearchProviderViewModel.js

```diff
@@ -72,11 +104,52 @@ CatalogItemNameSearchProviderViewModel.prototype.search = function(searchText) {
     });
 };
 
+function itemMatchesFilters(item, searchFilters) {
+    if (searchFilters.typeIs.length > 0 && searchFilters.typeIs.indexOf(item.type.toLowerCase()) < 0) {
+        return false;
+    }
+    if (searchFilters.typeIsNot.length > 0 && searchFilters.typeIsNot.indexOf(item.type.toLowerCase()) >= 0) {
+        return false;
+    }
+    if (!item.url) {
+        // if no URL, it can't match any positive or negative filters.
+        return searchFilters.urlMatches.length >= 0;
```

Comment:
I think this logic is wrong.  It should return false when the item has no URL and there's a URL filter, not true, right?